changelog.d/11950.removal

```diff
@@ -0,0 +1 @@
+Remove deprecated `user_may_create_room_with_invites` spam checker callback.
```

Comment:
It looks like the upgrade notes mention a replacement for this. Should we update `UPGRADE.md` again with this information (or include it here or mention to see those notes or something else)?

Reply:
Yeah that sounds like a good idea.